pkg/services/ngalert/state/persist.go

```diff
@@ -19,3 +22,8 @@ type InstanceStore interface {
 type RuleReader interface {
 	ListAlertRules(ctx context.Context, query *models.ListAlertRulesQuery) error
 }
+
+// Historian maintains an audit log of alert state history.
+type Historian interface {
+	RecordState(ctx context.Context, rule *ngmodels.AlertRule, labels data.Labels, evaluatedAt time.Time, currentData, previousData InstanceStateAndReason)
```

Comment:
The args to `RecordState` will definitely change in future PRs. I just kept the same args from the old `annotateState` helper for the time being. This was done to not over-complicate the diff of this PR.

I know the args aren't very elegant right now, the important bit is the abstraction and naming.